lumen/ai/ui.py

```diff
@@ -142,7 +143,8 @@ def _resolve_data(self, data: DataT | list[DataT] | None):
                 sources.append(src)
             elif isinstance(src, Pipeline):
                 mirrors[src.name] = src
-            elif isinstance(src, str):
+            elif isinstance(src, (str, Path)):
```

Comment:
if it is already a string, you don't need to reconvert it to one

```suggestion
            elif isinstance(src, Path):
```

Reply:
Calling str(src) is for converting the Path to string so we don't have to separate out the rest of the clause:

```python
                if src.startswith('http'):
                    remote = True
...
```